website/static/js/projectorganizer.js

```diff
@@ -1199,7 +867,470 @@ function _cleanupMithril() {
+                if (buttonEnabled) {
+                    $('#add-link-button').click(); //submits if the control is active
+                }
+            } else {
+                $('#add-link-warning').text('');
+                $('#add-link-button').addClass('tb-disabled');
+                linkName = '';
+                linkID = '';
+            }
+        });
+        $('#addprojectInput').bind('typeahead:selected', function (obj, datum, name) {
+            var getChildrenURL = theItem.apiURL + 'get_folder_pointers/',
+                children;
+            $.getJSON(getChildrenURL, function (data) {
+                children = data;
+                if (children.indexOf(datum.node_id) === -1) {
+                    $('#add-link-button').removeClass('tb-disabled');
+                    linkName = datum.name;
+                    linkID = datum.node_id;
+                } else {
+                    $('#add-link-warning').text('This project is already in the folder');
+                }
+            }).fail($osf.handleJSONError);
+        });
+    }
+
+}
+
+function addProjectEvent() {
+    var tb = this;
+    var item = tb.multiselected()[0];
+    var theItem = item.data;
+    var url = '/api/v1/pointer/',
+        postData = JSON.stringify({
+            pointerID: linkID,
+            toNodeID: theItem.node_id
+        });
+    theItem.expand = false;
+    saveExpandState(theItem, function () {
+        var postAction = $.ajax({
+            type: 'POST',
+            url: url,
+            data: postData,
+            contentType: 'application/json',
+            dataType: 'json'
+        });
+        postAction.done(function () {
+            tb.updateFolder(null, item);
+        });
+    });
+    triggerClickOnItem.call(tb, item);
+    tb.toolbarMode('bar');
+}
+
+function showLegend() {
+    var tb = this;
+    var keys = Object.keys(projectOrganizerCategories);
+    var data = keys.map(function (key) {
+        return {
+            icon: iconmap.componentIcons[key] || iconmap.projectIcons[key],
+            label: nodeCategories[key] || projectOrganizerCategories[key]
+        };
+    });
+    var repr = function (item) {
+        return [
+            m('span', {
+                className: item.icon
+            }),
+            '  ',
+            item.label
+        ];
+    };
+    var opts = {
+        footer: m('span', ['*lighter color denotes a registration (e.g. ',
+            m('span', {
+                className: iconmap.componentIcons.data + ' po-icon'
+            }),
+            ' becomes  ',
+            m('span', {
+                className: iconmap.componentIcons.data + ' po-icon-registered'
+            }),
+            ' )'
+            ])
+    };
+    tb.modal.update(legendView(data, repr, opts));
+    tb.modal.show();
+}
+
+var POToolbar = {
+    controller: function (args) {
+        var self = this;
+        self.tb = args.treebeard;
+        self.tb.toolbarMode = m.prop('bar');
```

Comment:
It might make sense to make the available modes constants. You can define that on the module-level or on POToolbar

```js
var POToolbar = {
    modes: {
        BAR: 'bar',
        SEARCH: 'search',
        //...
    },
    controller: function(args) {
        self.tb.toolbarMode = m.prop(self.modes.BAR); 
    }
}
```